src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/AvoidDoubleBraceInitializationCheck.java

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Detects double brace initialization.
+ * </p>
+ * <p>
+ * Rationale: Double brace initialization may look cool, but it is considered
+ * as anti-pattern and should be avoided.
+ * This is also can lead to a hard-to-detect memory leaks, since created anonymous class
+ * is not static, it holds an implicit reference to the outer class instance.
+ * See this
+ * <a href="https://blog.jooq.org/2014/12/08/dont-be-clever-the-double-curly-braces-anti-pattern/">
+ * blog post</a> and
+ * <a href="https://www.baeldung.com/java-double-brace-initialization">
+ * article</a> for more details.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;AvoidDoubleBraceInitialization&quot;/&gt;
+ * </pre>
+ * <p>
+ * Which results in the following violations:
+ * </p>
+ * <pre>
+ * class MyClass {
+ *   List list1 = new ArrayList&lt;Object&gt; { // violation
+ *     {
+ *       add(new Object());
+ *     }
+ *   };
+ *   List list2 = new ArrayList&lt;Object&gt; { // violation, comments are alone commas are ignored
```

Comment:
> comments are alone commas are ignored

I'm not sure I understand this phrase, also I don't see a comma. Was this suppose to be semicolon?

Reply:
corrected